sdk/openai/openai/test/public/embeddings.spec.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { matrix } from "@azure-tools/test-utils";
+import { describe, it, beforeAll } from "vitest";
+import { createClient } from "./utils/createClient.js";
+import {
+  APIMatrix,
+  APIVersion,
+  DeploymentInfo,
+  getDeployments,
+  withDeployments,
+} from "./utils/utils.js";
+import OpenAI, { AzureOpenAI } from "openai";
+import { assertEmbeddings, assertOpenAiError } from "./utils/asserts.js";
+import { embeddingModels } from "./utils/models.js";
+
+describe("Embeddings", function () {
+  matrix([APIMatrix] as const, async function (apiVersion: APIVersion) {
+    describe(`[${apiVersion}] Client`, () => {
+      let client: AzureOpenAI | OpenAI;
+      let deployments: DeploymentInfo[] = [];
+
+      beforeAll(async function () {
+        client = createClient(apiVersion, "completions");
+        deployments = await getDeployments("completions");
+      });
+
+      describe("getEmbeddings", function () {
+        it("embeddings test", async function () {
+          const prompt = ["This is text to be embedded"];
+          await withDeployments(
+            deployments,
+            (deploymentName) => client.embeddings.create({ model: deploymentName, input: prompt }),
+            assertEmbeddings,
+            embeddingModels,
+          );
+        });
+
+        it("wrong prompt type", async function () {
+          await assertOpenAiError(
+            client.embeddings.create({ model: "text-embedding-3-small", input: true as any }),
+            {
+              messagePattern: /'\$\.input' is invalid/,
+              type: `invalid_request_error`,
+              errorCode: null,
+            },
+          );
+        });
```

Comment:
I think we don't need to worry about cases where input arguments don't  match parameters types, perhaps we can delete this test.